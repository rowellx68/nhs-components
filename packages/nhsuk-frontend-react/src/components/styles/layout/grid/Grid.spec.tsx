import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  FullWidth as FullWidthStory,
  OneHalf as oneHalfStory,
  OneThird as OneThirdStory,
  TwoThirds as TwoThirdsStory,
  OneQuarter as OneQuarterStory,
  ThreeQuarters as ThreeQuartersStory,
  NestedGrid as NestedGridStory,
} from './Grid.stories';

const FullWidth = composeStory(FullWidthStory, meta);
const OneHalf = composeStory(oneHalfStory, meta);
const OneThird = composeStory(OneThirdStory, meta);
const TwoThirds = composeStory(TwoThirdsStory, meta);
const OneQuarter = composeStory(OneQuarterStory, meta);
const ThreeQuarters = composeStory(ThreeQuartersStory, meta);
const NestedGrid = composeStory(NestedGridStory, meta);

it('should render the FullWidth Grid component', () => {
  const { container } = render(<FullWidth />);

  expect(
    container.querySelector('.nhsuk-grid-column-full'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the OneHalf Grid component', () => {
  const { container } = render(<OneHalf />);

  expect(
    container.querySelector('.nhsuk-grid-column-one-half'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the OneThird Grid component', () => {
  const { container } = render(<OneThird />);

  expect(
    container.querySelector('.nhsuk-grid-column-one-third'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the TwoThirds Grid component', () => {
  const { container } = render(<TwoThirds />);

  expect(
    container.querySelector('.nhsuk-grid-column-two-thirds'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the OneQuarter Grid component', () => {
  const { container } = render(<OneQuarter />);

  expect(
    container.querySelector('.nhsuk-grid-column-one-quarter'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the ThreeQuarters Grid component', () => {
  const { container } = render(<ThreeQuarters />);

  expect(
    container.querySelector('.nhsuk-grid-column-three-quarters'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render the NestedGrid Grid component', () => {
  const { container } = render(<NestedGrid />);

  expect(
    container.querySelector('.nhsuk-grid-column-one-half'),
  ).toBeInTheDocument();
  expect(
    container.querySelector('.nhsuk-grid-column-two-thirds'),
  ).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
