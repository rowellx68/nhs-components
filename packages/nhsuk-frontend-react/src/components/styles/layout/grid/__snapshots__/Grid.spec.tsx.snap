// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the FullWidth Grid component 1`] = `
<div>
  <style>
    [class^="nhsuk-grid-column"] &gt; p {
            background-color: #005eb8;
            color: #fff;
            padding: 20px;
            text-align: center;
          }
  </style>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-full"
    >
      <p>
        nhsuk-grid-column-full
      </p>
    </div>
  </div>
</div>
`;

exports[`should render the NestedGrid Grid component 1`] = `
<div>
  <style>
    [class^="nhsuk-grid-column"] &gt; p {
            background-color: #005eb8;
            color: #fff;
            padding: 20px;
            text-align: center;
          }
  </style>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-two-thirds"
    >
      <p>
        nhsuk-grid-column-two-thirds
      </p>
      <div
        class="nhsuk-grid-row"
      >
        <div
          class="nhsuk-grid-column-one-half"
        >
          <p>
            nhsuk-grid-column-one-half
          </p>
        </div>
        <div
          class="nhsuk-grid-column-one-half"
        >
          <p>
            nhsuk-grid-column-one-half
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render the OneHalf Grid component 1`] = `
<div>
  <style>
    [class^="nhsuk-grid-column"] &gt; p {
            background-color: #005eb8;
            color: #fff;
            padding: 20px;
            text-align: center;
          }
  </style>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-one-half"
    >
      <p>
        nhsuk-grid-column-one-half
      </p>
    </div>
  </div>
</div>
`;

exports[`should render the OneQuarter Grid component 1`] = `
<div>
  <style>
    [class^="nhsuk-grid-column"] &gt; p {
            background-color: #005eb8;
            color: #fff;
            padding: 20px;
            text-align: center;
          }
  </style>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-one-quarter"
    >
      <p>
        nhsuk-grid-column-one-quarter
      </p>
    </div>
  </div>
</div>
`;

exports[`should render the OneThird Grid component 1`] = `
<div>
  <style>
    [class^="nhsuk-grid-column"] &gt; p {
            background-color: #005eb8;
            color: #fff;
            padding: 20px;
            text-align: center;
          }
  </style>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-one-third"
    >
      <p>
        nhsuk-grid-column-one-third
      </p>
    </div>
  </div>
</div>
`;

exports[`should render the ThreeQuarters Grid component 1`] = `
<div>
  <style>
    [class^="nhsuk-grid-column"] &gt; p {
            background-color: #005eb8;
            color: #fff;
            padding: 20px;
            text-align: center;
          }
  </style>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-three-quarters"
    >
      <p>
        nhsuk-grid-column-three-quarters
      </p>
    </div>
  </div>
</div>
`;

exports[`should render the TwoThirds Grid component 1`] = `
<div>
  <style>
    [class^="nhsuk-grid-column"] &gt; p {
            background-color: #005eb8;
            color: #fff;
            padding: 20px;
            text-align: center;
          }
  </style>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-two-thirds"
    >
      <p>
        nhsuk-grid-column-two-thirds
      </p>
    </div>
  </div>
</div>
`;
